src/main/java/com/microsoft/sqlserver/jdbc/TimeoutCommand.java

```diff
@@ -0,0 +1,36 @@
+package com.microsoft.sqlserver.jdbc;
+
+/**
+ * Abstract implementation of a command that can be timed out using the {@link TimeoutPoller}
+ */
+public abstract class TimeoutCommand<T> {
+    private final long startTime;
+    private final int timeout;
+    private final T command;
+    private final SQLServerConnection sqlServerConnection;
+
+    public TimeoutCommand(int timeout, T command, SQLServerConnection sqlServerConnection) {

```

Comment:
Please restrict visibility where ever possible. This PR shouldn't introduce any new public APIs. This also applies for TimePoller.java

Reply:
done